### DISCUSSION TITLE
nomenclature: `unit` vs `units`

### DISCUSSION THREAD

**Discussion by CagtayFabry** [BAMWelDX/weldx#528] [General]
Since we implemented our own `quantity` schema in #519 we no longer have to follow the asdf schema which would allow us to rename the `unit` field of the quantity

In general two options come to mind: `unit` oder `units`
I think it would be very good to decide on one of these options and use that consistently throughout `weldx`

Maybe we should look into other fields what is most common. I know that `pint-xarray` stores unit information under the `units` key but not sure if that is actually defined in the CF-conventions

Any preferences? please discuss

**Comment by vhirtham** [comment]
Not sure here but I would stick to the plural form -> `units`

**Comment by CagtayFabry** [comment]
a quick search for `unit:\n` gave the following results where we use that property name in schemas:
- quantity
- signal
- time_series
- variable

So I don't think it is too bad

if we want to rename to `units` we would have to provide the back compatibility for these schemas and we should certainly do that in the `0.5` release since otherwise the code will just get messier

I guess in these cases we only need a simple 
```python
if "units" in node:
  _units = U_(node["units"])
else:
  _units = U_(node["unit"])
```

**Reply by vhirtham** [reply]
I will turn this discussion into an issue and add it to the `0.5.0` milestone, okay?

**Reply by CagtayFabry** [reply]
I'll quickly draft the initial changes
